python_modules/dagster/dagster/_core/definitions/declarative_scheduling/dep_condition.py

```diff
@@ -0,0 +1,101 @@
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._serdes.serdes import whitelist_for_serdes
+
+from .asset_condition import AssetCondition, AssetConditionResult
+from .scheduling_condition_evaluation_context import SchedulingConditionEvaluationContext
+
+
+class DepConditionWrapperCondition(AssetCondition):
```

Comment:
Hmm at this point I'm a bit confused as to the relationship between `AssetCondition` and `SchedulingCondition` in this stack and going forward. Req'ing changes to we can discuss